v2/internal/testcommon/samples_tester.go

```diff
@@ -228,3 +239,46 @@ func IsSampleExcluded(path string, exclusions []string) bool {
 	}
 	return false
 }
+
+// updateARMReferencesForTest uses ReflectVisitor to visit through the ARMMetaObject to find and update ARMReferences
+func (t *SamplesTester) updateARMReferencesForTest(obj genruntime.ARMMetaObject) (genruntime.ARMMetaObject, error) {
+	visitor := reflecthelpers.NewReflectVisitor()
+	visitor.VisitStruct = t.setARMReference
+
+	err := visitor.Visit(obj, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "scanning for references of type %s", reflect.TypeOf(genruntime.ResourceReference{}))
+	}
+
+	return obj, nil
+}
+
+// setARMReference checks and sets the SubscriptionID and ResourceGroup name for ARM references to current values
+func (t *SamplesTester) setARMReference(this *reflecthelpers.ReflectVisitor, it reflect.Value, ctx interface{}) error {
+
+	if it.Type() != reflect.TypeOf(genruntime.ResourceReference{}) {
+		return reflecthelpers.IdentityVisitStruct(this, it, ctx)
+	}
+
+	if it.CanInterface() {
+		reference := it.Interface().(genruntime.ResourceReference)
+		if reference.ARMID != "" {
+			armIDField := it.FieldByName("ARMID")
+			if !armIDField.CanSet() {
+				return errors.New("cannot set 'ARMID' field of 'genruntime.ResourceReference'")
+			}
+
+			// Regex to match '/00000000-0000-0000-0000-000000000000/' strings, to replace with the subscriptionID
+			subMatcher := regexp.MustCompile("\\/([0]+-?)+\\/")
```

Comment:
minor: You could extract this regex so that we don't compile it every time? Could store it on the SamplesTester or just make it a package level variable:

`var subRegex = regex.MustCompile("\\/([0]+-?)+\\/")`